gcp-observability/src/test/java/io/grpc/gcp/observability/MetricsTest.java

```diff
@@ -0,0 +1,128 @@
+   */
+  @Ignore
+  @Test
+  public void testMetricsExporter() throws IOException, InterruptedException {
+    Sink mockSink = mock(GcpLogSink.class);
+    ObservabilityConfig mockConfig = mock(ObservabilityConfig.class);
+    InternalLoggingChannelInterceptor.Factory mockChannelInterceptorFactory =
+        mock(InternalLoggingChannelInterceptor.Factory.class);
+    InternalLoggingServerInterceptor.Factory mockServerInterceptorFactory =
+        mock(InternalLoggingServerInterceptor.Factory.class);
+
+    when(mockConfig.isEnableCloudMonitoring()).thenReturn(true);
+    when(mockConfig.getDestinationProjectId()).thenReturn(PROJECT_ID);
+
+    GcpObservability observability =
+        GcpObservability.grpcInit(
+            mockSink, mockConfig, mockChannelInterceptorFactory, mockServerInterceptorFactory);
+    observability.registerStackDriverExporter(PROJECT_ID, customTags);
+
+    Server server =
+        ServerBuilder.forPort(0)
+            .addService(new LoggingTestHelper.SimpleServiceImpl())
+            .build()
+            .start();
+    int port = cleanupRule.register(server).getPort();
+    SimpleServiceGrpc.SimpleServiceBlockingStub stub =
+        SimpleServiceGrpc.newBlockingStub(
+            cleanupRule.register(
+                ManagedChannelBuilder.forAddress("localhost", port).usePlaintext().build()));
+    assertThat(LoggingTestHelper.makeUnaryRpcViaClientStub("buddy", stub)).isEqualTo("Hello buddy");
+    // Adding sleep to ensure metrics are exported before querying cloud monitoring backend
+    TimeUnit.SECONDS.sleep(40);
+
+    // This checks Cloud monitoring for the new metrics that was just exported.
+    MetricServiceClient metricServiceClient = MetricServiceClient.create();
+    // Restrict time to last 1 minute
+    long startMillis = System.currentTimeMillis() - ((60 * 1) * 1000);
+    TimeInterval interval =
+        TimeInterval.newBuilder()
+            .setStartTime(Timestamps.fromMillis(startMillis))
+            .setEndTime(Timestamps.fromMillis(System.currentTimeMillis()))
+            .build();
+    // Timeseries data
+    String metricsFilter =
+        String.format(
+            "metric.type=\"custom.googleapis.com/opencensus/grpc.io/client/completed_rpcs\""
+                + " AND metric.labels.grpc_client_method=\"grpc.testing.SimpleService/UnaryRpc\""
+                + " AND metric.labels.%s=%s",
+            customTagKey, customTagValue);
+    ListTimeSeriesRequest metricsRequest =
+        ListTimeSeriesRequest.newBuilder()
+            .setName(ProjectName.of(PROJECT_ID).toString())
+            .setFilter(metricsFilter)
+            .setInterval(interval)
+            .build();
+    ListTimeSeriesPagedResponse response = metricServiceClient.listTimeSeries(metricsRequest);
+    assertThat(response.iterateAll()).isNotEmpty();
+    for (TimeSeries ts : response.iterateAll()) {
+      assertThat(ts.getPoints(0).getValue().getInt64Value()).isEqualTo(1);
```

Comment:
Can we also assert the method-name value (as one of the attached tags)?

Reply:
Done